### PR TITLE
Remove unused util CreatePods

### PR DESCRIPTION
**Pull request by alculquicondor**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

Method `CreatePods` was unused. Removed, but also renamed `CreatePodsWithControllerRef` to simply `CreatePods`.

#### Which issue(s) this PR fixes:

Follow up to #101601

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

/sig apps

### CODE REVIEW

File: pkg/controller/controller_utils.go [hunk 1]
```diff
@@ -445,12 +445,10 @@ func (r RealControllerRevisionControl) PatchControllerRevision(namespace, name s
...
-	// CreatePodsWithControllerRef creates new pods according to the spec, and sets object as the pod's controller.
-	CreatePodsWithControllerRef(namespace string, template *v1.PodTemplateSpec, object runtime.Object, controllerRef *metav1.OwnerReference) error
-	// CreatePodsWithControllerRefAndGenerateName creates new pods according to the spec, sets object as the pod's controller and sets pod's generateName.
-	CreatePodsWithControllerRefAndGenerateName(namespace string, template *v1.PodTemplateSpec, object runtime.Object, controllerRef *metav1.OwnerReference, generateName string) error
+	// CreatePods creates new pods according to the spec, and sets object as the pod's controller.
+	CreatePods(namespace string, template *v1.PodTemplateSpec, object runtime.Object, controllerRef *metav1.OwnerReference) error
+	// CreatePodsWithGenerateName creates new pods according to the spec, sets object as the pod's controller and sets pod's generateName.
+	CreatePodsWithGenerateName(namespace string, template *v1.PodTemplateSpec, object runtime.Object, controllerRef *metav1.OwnerReference, generateName string) error
```
Comment (ravisantoshgudimetla) [inline pkg/controller/controller_utils.go:451]: nit: I'd have split this change into another commit for the sake of readability.

**Review by ravisantoshgudimetla — APPROVED** [review]
LGTM apart from nit. Feel free to ignore it.


/lgtm

**Review by soltysh — APPROVED** [review]
/triage accepted
/priority backlog
/lgtm
/approve
